api/zulip/__init__.py

```diff
@@ -290,16 +290,20 @@ def get_user_agent(self):
                 vendor_version=vendor_version,
                 )
 
-    def do_api_query(self, orig_request, url, method="POST", longpolling=False):
-        # type: (Mapping[str, Any], str, str, bool) -> Dict[str, Any]
+    def do_api_query(self, orig_request, url, files=None, method="POST", longpolling=False):
```

Comment:
(I'm guessing this is the cause of the `production` test failures; basically, I think your code fails when using any API request other than uploading files because you try to iterate through None.